test/dep_env/read_c.rs

```diff
@@ -0,0 +1,6 @@
+use std::env::var;
+
+fn main() {
+    assert!(var("DEP_X_A").is_err());
```

Comment:
This assertion also passes if I just cherry-pick the changes to `test` onto the `main` branch without your other changes, and `s/link_deps/deps/g` in the BUILD.bazel file.

I _think_ one of the core observable pieces you're trying to fix in this PR is that if there's a _transitive_ dep on `:set_a_build`, this variable _still_ isn't set. I think you could demonstrate this by adding a `link_deps` dependency from `read_c` onto `:set_a` and a `deps` dependenecy from `read_c` onto `:set_a_build` - does that sound right to you? If so, could you add those to the BUILD.bazel file so that we have tests for this?

Reply:
Yeah, I am aware that it passes on master too. I just added the test as I felt that it was useful (I guess it probably should go in a separate PR). You may be correct that I didn't write the test correctly (whoops), but I think even your suggested test will pass in master.

I will investigate today, or tomorrow further and confirm if what I wrote above is correct.

Reply:
Yeah, it was working as I intended, but I see how you are correct that I could improve the test. I have done so.